gprofiler/metadata/application_identifiers.py

```diff
@@ -145,6 +145,53 @@ def get_app_id(self, process: Process) -> Optional[str]:
         return None
 
 
+class _UvicornApplicationIdentifierBase(_ApplicationIdentifier):
+    def uvicorn_to_app_id(self, wsgi_app_spec: str, process: Process) -> str:
+        wsgi_app_file = wsgi_app_spec.split(":", maxsplit=1)[0]
+        return f"uvicorn: {wsgi_app_spec} ({_append_python_module_to_proc_wd(process, wsgi_app_file)})"
+
+    def uvicorn_get_app_name(self, cmdline_list: List[str]) -> str:
+        # factory argument is being checked explicitly, because it contains minuses and precedes appid
+        cmdline_colon_list = list(filter(lambda x: ":" in x, cmdline_list))
+        for index, arg in enumerate(cmdline_list):
+            for colon_arg in cmdline_colon_list:
+                if arg == colon_arg and ("-" not in cmdline_list[index - 1] or cmdline_list[index - 1] == "--factory"):
+                    return arg
+        return ""
+
+
+class _UvicornApplicationIdentifier(_UvicornApplicationIdentifierBase):
+    def get_app_id(self, process: Process) -> Optional[str]:
+        # uvicorn seems to have similar ruling when it comes to naming as gunicorn
+
+        if "uvicorn" != os.path.basename(_get_cli_arg_by_index(process.cmdline(), 0)) and "uvicorn" != os.path.basename(
+            _get_cli_arg_by_index(process.cmdline(), 1)
+        ):
+            return None
+
+        # app specification does not have to be last, thus a method to extract it from commandline
+        return self.uvicorn_to_app_id(self.uvicorn_get_app_name(process.cmdline()), process)
+
+
+class _UvicornTitleApplicationIdentifier(_UvicornApplicationIdentifierBase):
```

Comment:
Is it relevant to uvicorn as well? I mean, usage of setproctitle, like gunicorn does.

Reply:
Need to check, will check :)

Reply:
Checked, no such method, removed the title identifier.